Assignment/palindrome_test.c

```diff
@@ -36,41 +35,29 @@ int IsValid (const char* num) {
    return 1;
 }
 
-void PrintRes (const char* inputs, const char* expOutputs, const char* actualRes, const char* res) { // Print the result
-   printf ("| %-30s | %-20s | %-20s | %-27s |\n", inputs, expOutputs, actualRes, res);
-}
-
 void TestCases () {
    char* inputs[] = { "vini", "I did,did I", "Ma*)d(*am", "@@@##!!!", "Mam!",
                       "Was it a car or a cat I saw?", "Don't nod", "Hello",
-                      "4334", "654", "34543", "-121", "*-23" ,"?@343", "6", " " };
-   char* expOutputs[] = { "Not a Palindrome", "Palindrome", "Palindrome", "Invalid Input",
-                          "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome",
-                          "Palindrome", "Not a Palindrome", "Palindrome", "Not a Palindrome",
-                          "Invalid Input", "Invalid Input", "Palindrome", "Invalid Input" };
+                      "4334", "2147447412", "34543", "-121", "*-23" ,"?@343", "6", " " },
+      * expOutputs[] = { "Not a Palindrome", "Palindrome", "Palindrome", "Invalid Input",
+                             "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome",
+                             "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome",
+                             "Invalid Input", "Invalid Input", "Palindrome", "Invalid Input" };
    int numOfTestCases = sizeof (inputs) / sizeof (inputs[0]);
    printf ("\n-------Testcases for String and Integer Input------\n\n"
            "| %-30s | %-20s | %-20s | %-10s |\n"
            "|--------------------------------|----------------------|"
            "----------------------|------------------|\n",
            "Input", "Expected Output", "Actual Output", "Test Case Result");
    for (int i = 0; i < numOfTestCases; i++) {
-      int actualOutput;
-      const char* actualRes;
-      if (IsValid (inputs[i])) {
-         int num = atoi (inputs[i]);
-         int reversed = 0;
-         actualOutput = ReverseNumber (num, &reversed);
-         actualRes = actualOutput == 1 ? "Palindrome" : "Not a Palindrome";
-      }
-      else {
-         actualOutput = PalindromeCheck (inputs[i]);
-         actualRes = actualOutput == 1 ? "Palindrome"
-            : actualOutput == 0 ? "Not a Palindrome" : "Invalid Input";
-      }
-      const char* res = (strcmp (actualRes, expOutputs[i]) == 0) ?
+      int actualOutput, reversed = 0;
+      actualOutput = IsValid (inputs[i]) ? ReverseNumber (atoi (inputs[i]), &reversed),

```

Comment:
The inputs[i] is used several times inside the loop. Each call accesses the memory location of the array and gets the element of the particular index. Sometimes it may cache the value as it has several calls. But when there is a large number of test cases the performance can reduce. Instead, you can store the input in a local variable inside the for loop and use it.